carla/recourse_methods/catalog/cem/model.py

```diff
@@ -142,35 +93,28 @@ def __init__(self, sess, catalog_model: MLModel, hyperparams):
             ImgToEnforceLabel_Score_s
         )
 
-        Loss_Attack = self.__compute_attack_loss(
-            self.max_nontarget_lab_score, self.target_lab_score, self.mode
-        )
-        Loss_Attack_s = self.__compute_attack_loss(
-            max_nontarget_lab_score_s, target_lab_score_s, self.mode
-        )
-
         # sum up the losses
         self.Loss_L1Dist = tf.reduce_sum(L1_dist)
-
-        self.Loss_L2Dist = tf.reduce_sum(L2_dist)
-        Loss_L2Dist_s = tf.reduce_sum(L2_dist_s)
-
-        self.Loss_Attack = tf.reduce_sum(self.const * Loss_Attack)
-        Loss_Attack_s = tf.reduce_sum(self.const * Loss_Attack_s)
-
-        delta_input_ae = delta_img if self.mode == "PP" else self.adv_img
-        delta_input_ae_s = delta_img_s if self.mode == "PP" else self.adv_img_s
-
-        self.Loss_AE_Dist = self.__compute_AE_lost(delta_input_ae)
-        Loss_AE_Dist_s = self.__compute_AE_lost(delta_input_ae_s)
-
-        Loss_ToOptimize = Loss_Attack_s + Loss_L2Dist_s + Loss_AE_Dist_s
-        self.Loss_Overall = (
-            self.Loss_Attack
-            + self.Loss_L2Dist
-            + self.Loss_AE_Dist
-            + tf.multiply(beta, self.Loss_L1Dist)
+        (
+            self.Loss_L2Dist,
+            self.Loss_Attack,
+            self.Loss_AE_Dist,
+            Loss_ToOptimize,
+        ) = self.__compute_losses(
+            delta_img,
+            self.adv_img,
+            L2_dist,
+            self.max_nontarget_lab_score,
+            self.target_lab_score,
         )
+        (_, _, _, Loss_ToOptimize_s,) = self.__compute_losses(
```

Comment:
Its a bit funny to have a function to compures 4-5 losses, but for some cases we only need one return variable. Is there a way to better structure the functions?